code/game/turfs/simulated/floor.dm

```diff
@@ -276,6 +293,15 @@ turf/simulated/floor/proc/update_icon()
 	else if(is_grass_floor())
 		src.icon_state = "sand[pick("1","2","3")]"
 		broken = 1
+	else if(is_mineral_floor())
+		if(material=="diamond") return //diamond doesn't break
+		if(material=="phazon") //Phazon shatters

```

Comment:
1. why aren't these just using the material datum for the material
2. why are these snowflaked in and why aren't they procs on said material datum